### ISSUE TITLE
SAE IBP-MAP K=2 torus recovery: joint fit collapses the assignment off both true atoms (mean active mass 0.0297 << 0.2) despite an LSQ-seeded truth decoder

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#853]
## Summary

`src/terms/sae_manifold.rs::tests::ibp_map_k2_periodic_torus_recovers_signal_with_lsq_init` (`sae_manifold.rs:11955`) fails its anti-collapse assertion:

```
mean active mass across rows = 0.0297 should exceed 0.2; assignment did not collapse
```

Reproduced on committed main in Rust CI run 27084917256.

## What the test exercises

A synthetic K=2 torus signal `Z = [cos θ1, sin θ1, cos θ2, sin θ2] @ mix` (`sae_manifold.rs:11964`), atoms initialised at the (shifted) true angles with an **LSQ-seeded decoder** (joint normal-equations solve, `sae_manifold.rs:12027`), IBP-MAP assignment (`AssignmentMode::ibp_map(0.7, 1.0, false)`, `sae_manifold.rs:12079`), then 30 outer iterations of `run_joint_fit_arrow_schur` (`sae_manifold.rs:12086`). It asserts both an R² recovery (`> 0.5`) **and** that the assignment did not collapse:

```rust
let assignments = term.assignment.assignments();
let mean_active: f64 = assignments.iter().copied().sum::<f64>() / (n as f64);
assert!(mean_active > 0.2, "mean active mass ... = {mean_active:.4} should exceed 0.2; assignment did not collapse");
```

The observed mean active mass `0.0297` (vs the `> 0.2` floor) means the IBP gate `σ(logit/τ)·π_k` was driven to ≈ 0 across rows during the joint fit — i.e. the assignment **collapsed off** even though both atoms carry real signal and the decoder was LSQ-seeded to the truth.

## Root cause area (file:line)

The collapse happens inside the joint `(t, β, logit)` Newton fit, which descends the IBP assignment-sparsity prior. The IBP logit gradient/Hessian enters the per-row block via `assignment_prior_grad_hdiag` (`AssignmentMode::IBPMap` branch, `sae_manifold.rs:8484`) → `IBPAssignmentPenalty`, and the logit JVP rows feed the data-fit coupling (`fill_assignment_logit_jvp_rows`, `sae_manifold.rs:8086`-region). The geometric stick-breaking shrinkage schedule `π_k = (α/(α+1))^k` (`sae_manifold.rs:8136`) with `α = 1.0` already halves atom-1's prior mass; combined with the sparsity-prior gradient pulling logits negative, the gate is driven toward 0 faster than the data-fit term can hold it active. The `lambda_sparse = exp(0.0) = 1` (`SaeManifoldRho::new(0.0, ...)`, `sae_manifold.rs:12083`) gives the sparsity prior full weight against the LSQ-seeded reconstruction.

## Discrimination

Unlike the other SAE failures in this CI run, this is **not** a constructor/contract/tolerance mismatch: the term constructs and the 30 iterations run to completion. It is a genuine quality failure — the IBP-MAP joint fit collapses the assignment off both true atoms despite a truth-seeded decoder, so the gate dynamics (IBP sparsity-prior gradient vs data-fit JVP support) over-shrink at the configured `lambda_sparse`/`alpha`/`temperature`. This is the manifold-SAE analogue of the convergence/over-shrinkage failures tracked in #795 (arrow-Schur ridge saturation on a planted circle). Needs investigation of the IBP gate gradient balance in `run_joint_fit_arrow_schur` at the test's hyperparameters; if the collapse is a true optimum at these settings, the test's anti-collapse floor or the prior weighting is the lever.
